identity/identity.go

```diff
@@ -36,97 +33,105 @@ func (s *StateStorage) Set(
 	publishEpoch *types.EpochID,
 	newState State,
 ) {
-	s.mu.Lock()
-	defer s.mu.Unlock()
-
-	if _, exists := s.identities[id]; !exists {
-		s.identities[id] = []StateInfo{}
-	}
-
-	if len(s.identities[id]) > 100 {
-		s.identities[id] = s.identities[id][1:]
-	}
-
 	info := StateInfo{
 		State:        newState,
 		PublishEpoch: publishEpoch,
 		Time:         time.Now(),
 	}
 
-	s.identities[id] = append(s.identities[id], info)
+	stateBytes, err := marshalState(&info)
+	if err != nil {
+		panic(fmt.Sprintf("marhsaling state: %v", err))
```

Comment:
```suggestion
		panic(fmt.Sprintf("marshaling state: %v", err))
```

I'm not a big fan of just panicing here. How about adding a logger to `StateStorage` and at least `log.Fatal` or `log.Panic` instead? 🙂